app/services/user_stat_count_updater.rb

```diff
@@ -22,6 +22,18 @@ def update!(post, user_stat: nil, action: :increment!)
       elsif post.post_type == Post.types[:regular]
         stat.public_send(action, :post_count)
       end
+    rescue ActiveRecord::StatementInvalid => e
+      if e.cause.is_a?(PG::CheckViolation)
```

Comment:
Will there ever be a case where `e.cause` is nil?

Reply:
It can be `nil` but it'll respond to `is_a?` since it is an `Object`